torch/distributed/_tensor/ops/embedding_ops.py

```diff
@@ -1,97 +1,168 @@
 # Copyright (c) Meta Platforms, Inc. and affiliates
 # implement matrix related ops for distributed tensor
+import itertools
+from typing import cast, List
 
 import torch
-from torch.distributed._tensor.op_schema import OpSchema, OutputSharding
-from torch.distributed._tensor.ops.utils import register_prop_rule
+from torch.distributed._tensor.op_schema import (
+    OpSchema,
+    OpStrategy,
+    PlacementStrategy,
+    StrategyType,
+)
+from torch.distributed._tensor.ops.utils import (
+    generate_redistribute_costs,
+    is_tensor_shardable,
+    register_op_strategy,
+)
 
 from torch.distributed._tensor.placement_types import (
     _Partial,
     DTensorSpec,
+    Placement,
     Replicate,
     Shard,
 )
 
+from torch.distributed.device_mesh import DeviceMesh
+
 aten = torch.ops.aten
 
 
-# TODO: Enable BWD for embedding op.
-@register_prop_rule(aten.embedding.default)
-def embedding_rules(op_schema: OpSchema) -> OutputSharding:
-    weight_spec, inp_spec = op_schema.args_spec
+@register_op_strategy(aten.embedding.default)
+def embedding_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> StrategyType:
+    """
+    This strategy handles embedding op. We have two possible embedding shardings:
+    rowwise and colwise
+    # TODO: implement rowwise sharding
+    """
+    weight_strategy = cast(OpStrategy, op_schema.args_schema[0])
+    indices_strategy = cast(OpStrategy, op_schema.args_schema[1])
+
+    weight_shape = weight_strategy.output_shape
+    indices_shape = indices_strategy.output_shape
+    output_emd_dim = len(indices_shape)
+
+    # guard rowwise sharding not implemented for now
+    weight_spec = weight_strategy.strategies[0].out_spec
```

Comment:
since #116437 is being merged, replace `out_spec` with `output_spec`.

Reply:
Yeah If you review the next PR, I removed this code piece in the next PR so it would not trigger failures if we land things together. I'll try to do a rebase just to make sure